src/app/loadout-builder/filter/ModPicker.tsx

```diff
@@ -144,165 +145,153 @@ function mapStateToProps() {
   });
 }
 
-interface State {
-  query: string;
-  height?: number;
-  lockedArmor2Mods: LockedArmor2ModMap;
-}
-
 /**
  * A sheet that allows picking a perk.
  */
-class ModPicker extends React.Component<Props, State> {
-  state: State = {
-    query: '',
-    lockedArmor2Mods: copy(this.props.lockedArmor2Mods),
-  };
-  private itemContainer = React.createRef<HTMLDivElement>();
-  private filterInput = React.createRef<SearchFilterRef>();
-
-  componentDidMount() {
-    if (this.itemContainer.current) {
```

Comment:
One of the few actual changes I did was remove this altogether and just set a max height using a vh value. TBH I am not quite sure why this was needed over a max height?

Reply:
We set the height on first render because there's a search field, and if the height isn't set when you search the popup will resize smaller. I plan on building this feature into the Sheet itself, but in the meantime please leave it in place. https://github.com/DestinyItemManager/DIM/issues/5583

Reply:
Hmm ok, issue I have is with the functional component it gets set to a size of 16px on first render rather than something reasonable. I will have a play to see if I can get it working but if not maybe this isn't possible without moving it to the sheet.

Reply:
That makes it sound like you're not using `useLayoutEffect`

Reply:
I figured out what I was doing wrong...I may be a bit special...I will leave it at that.